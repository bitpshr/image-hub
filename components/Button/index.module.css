@import "~lib/variables.css";

.button {
	background: var(--primary);
	border-radius: calc(var(--grid-size) / 4);
	border: 1px solid var(--primary);
	color: var(--white);
	cursor: pointer;
	font: inherit;
	letter-spacing: calc(var(--grid-size) / 4);
	padding: var(--grid-size) calc(var(--grid-size) * 2);
	text-transform: uppercase;
}

.small {
	font-size: var(--font-size-small);
	padding: calc(var(--grid-size) / 2) var(--grid-size);
}
