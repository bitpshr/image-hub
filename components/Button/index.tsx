import { ButtonHTMLAttributes, DetailedHTMLProps, memo } from "react";

import css from "./index.module.css";

export interface ButtonProps
	extends DetailedHTMLProps<
		ButtonHTMLAttributes<HTMLButtonElement>,
		HTMLButtonElement
	> {
	size?: "small" | "normal";
}

const Button = memo<ButtonProps>((props) => (
	<button
		{...props}
		className={`${css.button} ${props.size === "small" ? css.small : ""}`}
	/>
));

export default Button;
