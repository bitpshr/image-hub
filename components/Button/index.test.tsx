import React from "react";
import renderer from "react-test-renderer";

import Button from "./";

test("default render", () => {
	const component = renderer.create(<Button size="normal" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("normal size", () => {
	const component = renderer.create(<Button size="normal" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("small size", () => {
	const component = renderer.create(<Button size="small" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
