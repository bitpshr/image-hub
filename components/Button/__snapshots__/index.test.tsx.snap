// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<button
  className="button "
  size="normal"
/>
`;

exports[`normal size 1`] = `
<button
  className="button "
  size="normal"
/>
`;

exports[`small size 1`] = `
<button
  className="button small"
  size="small"
/>
`;
